packages/ui/components/select-rich/test/lion-select-rich.test.js

```diff
@@ -329,6 +329,21 @@ describe('lion-select-rich', () => {
       expect(elSingleoption.opened).to.be.false;
     });
 
+    it('stays closed with [ArrowUp] or [ArrowDown] key in readonly mode', async () => {
+      const elReadOnly = await fixture(html`
+        <lion-select-rich readonly>
+          <lion-option .choiceValue=${10}>Item 1</lion-option>
+          <lion-option .choiceValue=${20} checked>Item 2</lion-option>
+        </lion-select-rich>
+      `);
+
+      elReadOnly.dispatchEvent(new KeyboardEvent('keyup', { key: 'ArrowDown' }));
+      expect(elReadOnly.opened).to.be.false;
```

Comment:
Fix looks good to me!

I just want to share my finding that `elReadOnly.opened` will be `false` even without `readonly` attribute because `opened` is updated asynchronously.

Adding `await aTimeout(500)` between dispatching event and the `expect(...)` will do the job, although introducing timeout is not ideal. If you have a better idea, go for it!

The test for disabled has the same problem which was addressed in https://github.com/ing-bank/lion/pull/2237.

Reply:
Before adding my code changes, I ensured the test failed, proving its validity. It passed only after applying my changes. I also removed the `readonly` attribute to test further, as you suggested. The test failed in this scenario, indicating that `elReadOnly.opened` updates on time as expected.

Reply:
Sorry, indeed I cannot reproduce it. Sorry for the confusion!